tests/suites/test_suite_bignum_mod_raw.function

```diff
@@ -294,7 +294,73 @@ exit:
 /* END MERGE SLOT 6 */
 
 /* BEGIN MERGE SLOT 7 */
+/* BEGIN_CASE */
+void mpi_mod_raw_to_mont_rep( char * input_N, char * input_A, char * input_X )
+{
+    mbedtls_mpi_uint *N = NULL;
+    mbedtls_mpi_uint *A = NULL;
+    mbedtls_mpi_uint *X = NULL;
+    mbedtls_mpi_mod_modulus m;
+    size_t n_limbs, a_limbs, x_limbs;
+
+    /* Read inputs */
+    TEST_EQUAL( 0, mbedtls_test_read_mpi_core( &N, &n_limbs, input_N ) );
+    TEST_EQUAL( 0, mbedtls_test_read_mpi_core( &A, &a_limbs, input_A ) );
+    TEST_EQUAL( 0, mbedtls_test_read_mpi_core( &X, &x_limbs, input_X ) );
+
+    /* Test that input does not require more limbs than modulo */
+    TEST_LE_U(a_limbs, n_limbs);
+
+    mbedtls_mpi_mod_modulus_init( &m );
+    TEST_EQUAL( 0, mbedtls_mpi_mod_modulus_setup( &m, N, n_limbs,
+                MBEDTLS_MPI_MOD_EXT_REP_BE, MBEDTLS_MPI_MOD_REP_MONTGOMERY ) );
+
+    /* Convert from cannonical into Montgomery representation */
+    TEST_EQUAL(0, mbedtls_mpi_mod_raw_to_mont_rep( A, &m ) );
+
+    /* The result matches expected value */
+    ASSERT_COMPARE( A, x_limbs, X, x_limbs );
```

Comment:
must fix: `ASSERT_COMPARE()` takes bytes, not limbs (sorry, that's my fault for suggesting using it without being explicit enough about the limbs -> bytes change)

Reply:
Nah it is my fault for assuming it would be a drop in replacement. It has been addressed. 

The PR is being rebased since the precceeding pr has been updated so the change will be squashed into the `mbedtls_test_read_mpi_core` patch

Reply:
It might make sense to define an `ASSERT_COMPARE_MPI_CORE`, taking a single size expressed in limbs. A single size since it doesn't make sense to to a simple byte comparison otherwise: two numbers represented with different sizes can be equal, if the bigger one has only zeros in the high-order limbs. At that point you'd really need to call `mbedtls_mpi_cmp_xxx`.

Reply:
Yes, I agree, although I'd prefer to see the same start with `TEST_` :)